host/vm-manager/0001-Add-support-for-HPD.patch

```diff
@@ -0,0 +1,144 @@
++                "HDMI-1",
++                "HDMI-2",
++                "HDMI-3",
++                "HDMI-4",
++                "DP-1",
++                "DP-2",
++                "DP-3",
++                "DP-4",
++        };
++
+ static bool CheckUuid(std::string uuid) {
+     try {
+         boost::uuids::string_generator gen;
+@@ -281,6 +292,37 @@ bool VmBuilderQemu::SetupSriov(void) {
+         emul_cmd_.append(" -display gtk,gl=on,monitor=" + vgpu_mon_id);
+     }
+ 
++    int numberOfDisplays = 0;
++    std::string connectors_0 = cfg_.GetValue(kGroupDisplay, kDispConnector0);
++    std::string connectors_1 = cfg_.GetValue(kGroupDisplay, kDispConnector1);
++    std::string connector_str;
++    std::string show_fps = cfg_.GetValue(kGroupDisplay, kDispShowFps);
++    if (!show_fps.empty() && (show_fps.compare("on") == 0 || show_fps.compare("off") == 0)) {
++        emul_cmd_.append(",show-fps=" + show_fps + ",");
++    } else if (show_fps.empty()){
++        emul_cmd_.append(",show-fps=off,");
++    } else {
++	LOG(error) << "Please enter a valid value for show_fps.";
++	return false;
++    }
++
++    for(int i = 0; i < 8; i++) {
++        if(!connectors_0.empty() && connectors_0.compare(connectors[i]) == 0) {
++                        numberOfDisplays++;
++                        connector_str.append("connectors.0=");
++                        connector_str.append(connectors_0);
++                        connector_str.append(",");
++        }
++        if(!connectors_1.empty() && connectors_1.compare(connectors[i]) == 0) {
++                        numberOfDisplays++;
++                        connector_str.append("connectors.1=");
++                        connector_str.append(connectors_1);
++                        connector_str.append(",");
++        }
++    }
++
++    emul_cmd_.append(connector_str);
++
+     std::string mem_size = cfg_.GetValue(kGroupMem, kMemSize);
+     boost::trim(mem_size);
+ 
+@@ -292,7 +334,7 @@ bool VmBuilderQemu::SetupSriov(void) {
+     if (vf < 0)
+         return false;
+ 
+-    emul_cmd_.append(" -device virtio-vga,max_outputs=1,blob=true"
++    emul_cmd_.append(" -device virtio-vga,max_outputs=" + ((numberOfDisplays > 0 ) ? std::to_string(numberOfDisplays) : "1") + ",blob=true"
+                     " -device vfio-pci,host=0000:00:02." + std::to_string(vf) +
+                     " -object memory-backend-memfd,hugetlb=on,id=mem_sriov,size=" + mem_size +
+                     " -machine memory-backend=mem_sriov");
+@@ -791,12 +833,15 @@ void VmBuilderQemu::BuildVinputCmd(void) {
+ }
+ 
+ void VmBuilderQemu::BuildDispCmd(void) {
++    std::string vgpu_type = cfg_.GetValue(kGroupVgpu, kVgpuType);
++    if(vgpu_type.compare(kVgpuSriov) == 0) {
++	    return;
++    }
+     std::string disp_op = cfg_.GetValue(kGroupDisplay, kDispOptions);
+     if (disp_op.empty()) {
+         emul_cmd_.append(" -display gtk,gl=on");
+         return;
+     }
+-
+     emul_cmd_.append(" -display " + disp_op);
```

Comment:
Any idea why "-display" command is appended here in buildDispCmd() and also in SetupSriov()? 
Which one takes precedence?

Reply:
Yes, this is a bug from vm-manager side. I have already informed Yadong about the same. He will take care of it.